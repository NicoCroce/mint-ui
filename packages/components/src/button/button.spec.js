import * as React from "react";
import { cleanup, fireEvent, render } from "@testing-library/react";

import Button from "./index";

afterEach(cleanup);
describe("Button", () => {
	test("renders button", () => {
		const { asFragment } = render(<Button>Test</Button>);
		expect(asFragment()).toMatchSnapshot();
	});
	test("captures button click", () => {
		const handleClick = jest.fn();

		const { getByTestId } = render(
			<Button data-testid="capture-click" onClick={handleClick}>
				Capture click
			</Button>
		);
		const node = getByTestId("capture-click");
		fireEvent.click(node);
		expect(handleClick).toHaveBeenCalledTimes(1);
	});
});

afterEach(cleanup);
