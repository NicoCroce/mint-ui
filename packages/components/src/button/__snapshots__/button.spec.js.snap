// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders button 1`] = `
<DocumentFragment>
  <button
    class="button__StyledButton-sc-1o82sov-0 eNCDSg tm-button- "
    type="button"
  >
    <span
      style="vertical-align: middle; margin-left: 0px;"
    >
      Test
    </span>
  </button>
</DocumentFragment>
`;
